NebulaWorld/RemotePlayerModel.cs

```diff
@@ -32,6 +33,7 @@ public RemotePlayerModel(ushort playerId)
                 // Add remote player components
                 Movement = PlayerTransform.gameObject.AddComponent<RemotePlayerMovement>();
                 Animator = PlayerTransform.gameObject.AddComponent<RemotePlayerAnimation>();
+                Effects = PlayerTransform.gameObject.AddComponent<RemotePlayerEffects>();
```

Comment:
This is the reason for the race condition in `RemotePlayerMovement`, your `RemotePlayerEffects` was added after the `RemotePlayerMovement` script, so when the `RemotePlayerMovement.Awake()` was called, there was no `RemotePlayerEffects` on the player gameobject.

But the right fix is to change the `RemotePlayerMovement.Awake()` to `RemotePlayerMovement.Start()`